pinot-segment-local/src/main/java/org/apache/pinot/segment/local/io/writer/impl/FixedByteChunkSVForwardIndexWriter.java

```diff
@@ -101,6 +103,12 @@ public void putDouble(double value) {
     flushChunkIfNeeded();
   }
 
+  public void putBigDecimal(BigDecimal value) {
+    _chunkBuffer.put(BigDecimalUtils.serialize(value));
+    _chunkDataOffset += BigDecimalUtils.byteSize(value);
```

Comment:
Put the serialized bytes length here instead of re-calculating the byte size again